lib/matplotlib/axes/_axes.py

```diff
@@ -8094,3 +8094,13 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
     tricontourf = mtri.tricontourf
     tripcolor = mtri.tripcolor
     triplot = mtri.triplot
+
+    def _get_aspect_ratio(self):
+        """
+        Convenience method to calculate the aspect ratio of the axes in
+        the display coordinate system.
+        """
+        figure_size = self.get_figure().get_size_inches()
+        ll, ur = self.get_position() * figure_size
+        width, height = ur - ll
+        return height / width * self.get_data_ratio()
```

Comment:
```suggestion
        return height / (width * self.get_data_ratio())
```
See the `imshow()` example in my comments – as `self.get_data_ratio()` returns y/x in data range, division seems correct here; in any case the example plot looks properly rectangular with that change applied.
```